src/Perf/IIDOptimizer/Program.cs

```diff
@@ -76,13 +74,34 @@ private static int GuidPatch(string targetAssembly, string outputDirectory, IEnu
 
                 var targetAssemblyDefinition = AssemblyDefinition.ReadAssembly(targetAssembly, readerParameters);
 
+                /// Set WinRT.Runtime.dll -- either we are patching it, or it is in the references 
+                AssemblyDefinition? winRTRuntimeAssembly = null;
+
+                if (targetAssemblyDefinition.Name.Name == "WinRT.Runtime")
+                {
+                    winRTRuntimeAssembly = targetAssemblyDefinition;
+                }
+                else
+                {
+                    foreach (var referenceAssembly in targetAssemblyDefinition.MainModule.AssemblyReferences)
+                    {
+                        if (referenceAssembly.Name == "WinRT.Runtime")
+                        { 
+                            winRTRuntimeAssembly = resolver.Resolve(referenceAssembly);
+                        }
+                    }
+
+                }
+                /// 
+
+                /// Don't patch twice 
                 if (targetAssemblyDefinition.MainModule.Types.Any(typeDef => typeDef.Name == "<GuidPatcherImplementationDetails>"))
                 {
                     Console.WriteLine("Target assembly has already been patched. Exiting early as there is no work to do.");
                     return -2;
                 }
 
-                var guidPatcher = new GuidPatcher(winRTRuntimeAssembly, targetAssemblyDefinition);
+                var guidPatcher = new GuidPatcher(winRTRuntimeAssembly!, targetAssemblyDefinition);

```

Comment:
I think it would be worth handling the `winRTRuntimeAssembly == null` case explicitly like how the "don't patch twice" case is handled.